Woodoku/src/Entity/WoodokuBoard.py

```diff
@@ -25,89 +25,191 @@ def remove_blocks(self, blocks_coord: List[Tuple[int, int]]) -> None:
 
-    def __get_row_all_coords(self, row_num: int) -> List[Tuple[int, int]]:
-        pass
+        Terminology Reference:
+        [Wikipedia Sudoku Glossary](https://en.wikipedia.org/wiki/Glossary_of_Sudoku#Terminology_and_grid_layout)
 
-    def __get_column_all_coords(self, column_num: int) -> List[Tuple[int, int]]:
-        pass
+        Returns:
+            Tuple[Dict[str, int], Set[Tuple[int,int]]]:
+                1. A dictionary mapping from name of the group to the number of groups.
+                e.g. { "row": 1, "column": 2, "box": 0 }
+                2. Set of block coordinates for the group
 
-    def __get_3x3_all_coords(self, index: int) -> List[Tuple[int, int]]:
         """
-        The index of 3x3 block is as following:
-        _____________
-        | 0 | 1 | 2 |
-        | 3 | 4 | 5 |
-        | 6 | 7 | 8 |
-        _____________
-
-        :param index: the index as above
-        :return: All the coords in this 3x3 block.
+        info = {"row": 0, "column": 0, "box": 0}
+        rep = self.__representation
+        group_blocks = set()  # use set to handle overlapping group removal
+        for index in range(9):
+            row_blocks = self.__get_row_coords(index)
+            col_blocks = self.__get_col_coords(index)
+            box_blocks = self.__get_box_coords(index)
+
+            if rep.is_occupied(row_blocks):
+                info["row"] += 1
+                group_blocks.update(row_blocks)
+
+            if rep.is_occupied(col_blocks):
+                info["column"] += 1
+                group_blocks.update(col_blocks)
+
+            if rep.is_occupied(box_blocks):
+                info["box"] += 1
+                group_blocks.update(box_blocks)
+
+        return info, group_blocks
+
+    def __get_row_coords(self, row_index: int) -> List[Tuple[int, int]]:
+        return [(row_index, col) for col in range(9)]
+
+    def __get_col_coords(self, col_index: int) -> List[Tuple[int, int]]:
+        return [(row, col_index) for row in range(9)]
+
+    def __get_box_coords(self, index: int) -> List[Tuple[int, int]]:
+        """The index of 3x3 box is as following:
+        | 0 	| 1 	| 2 	|
+        |---	|---	|---	|
+        | 3 	| 4 	| 5 	|
+        | 6 	| 7 	| 8 	|
+
+        Args:
+            index (int): The index of 3x3 box as list above
+
+        Returns:
+            List[Tuple[int, int]]: All the coordinates in that 3x3 box
+
         """
-        pass
+        # map to 3x3 coordinates
+        # index -> 3x3 coordinates
+        #  0 -> 0,0
+        #  1 -> 0,1
+        #  2 -> 0,2
+        #  3 -> 1,0
+        #  4 -> 1,1
+        #  5 -> 1,2
+        #  6 -> 2,0
+        #  7 -> 2,1
+        #  8 -> 2,2
+        x, y = index // 3, index % 3
+
+        # map to 9x9 coordinates
+        x, y = x * 3, y * 3
+        return [(row, col) for row in range(x, x + 3) for col in range(y, y + 3)]
+    
+    def __str__(self) -> str:
+        return str(self.__representation)
```

Comment:
It should include score as part of `__str__` print out. Mainly use this method in the `CommandLineUI`.

But could leave this method blank for now.